moto/resourcegroupstaggingapi/models.py

```diff
@@ -265,6 +266,22 @@ def format_tag_keys(
                     continue
                 yield {"ResourceARN": bucket.arn, "Tags": tags}
 
+        # Cloud Directory
+        if not resource_type_filters or "clouddirectory" in resource_type_filters:
+            try:
+                clouddirectory_backend = clouddirectory_backends[self.account_id][
+                    self.region_name
+                ]
+                for directory in clouddirectory_backend.directories.values():
+                    tags = clouddirectory_backend.tagger.list_tags_for_resource(
+                        directory.directory_arn
+                    )["Tags"]
+                    if not tags or not tag_filter(tags):
+                        continue
+                    yield {"ResourceARN": f"{directory.directory_arn}", "Tags": tags}
+            except Exception:
+                pass
```

Comment:
What kind of exception can happen here, and why do we ignore it? Note that I don't mind - but it would be nice to have a comment explaining what's happening for future reference :slightly_smiling_face:

Reply:
Ah, that makes sense, if the `CloudDirectory` service is not available in that region.

It's probably cleaner to use the example from `LexV2` then, with a custom method that returns  `None` if no region exists. Then the whole section can be if-guarded instead, to make sure it only runs when `CloudDirectory` is available.